10/benchmark_tcp_pipeline.md

```diff
@@ -0,0 +1,77 @@
+# Benchmarking the TCP Connection between BDB and Tendermint
+
+#### Running the TCP Benchmark Tests
+
+In order to execute a TCP benchmark test, first install [tm-bench][tm-bench].
+
+We used a couple of handy bash scripts to [specify test parameters][tests] and [execute tests][runtests] to output test results in a nice format for a csv. You can download and try them yourself.
+
+We took [data] on hardware with the following specifications
+
+* Processor: Intel(R) Xeon(R) CPU E5-2650 v3 @ 2.30GHz (10 cores)
+* Memory: 15GB
+* OS: Ubuntu 18.04 LTS
+* Tendermint version: 0.19.5-747f28f8
+
+For the following graphs, we use the coloring scheme:
+* Blue
+  * KV-store integrated with Tendermint (no TCP pipeline)
+  * Logging enabled 
+* Red
+  * KV-store integrated with Tendermint (no TCP pipeline)
+  * Logging disabled
+* Yellow
+  * KV-store remote from Tendermint (yes TCP pipeline)
+  * Logging enabled
+* Green
+  * KV-store remote from Tendermint (yes TCP pipeline)
+  * Logging disabled
+
+Examining the data collected, we can see a few patterns.
+
+###### Transactions per Second / Duration
+![tx/s vs duration][duration]
+
+FIrst, notice that there is a severe degradation in performance under a sustained heavy load. The integrated processes (no TCP connection) initially slightly out perform the external processes, but eventually, they all reach a steady state of 210 tx/s.
```

Comment:
I suspect, degradation begins when the mempool becomes full. Can you set the `size_of_mempool` Tendermint config to `10` and see how transaction rate sustains then?